griptape/tools/rest_api_client/tool.py

```diff
@@ -55,9 +57,10 @@ def put(self, params: dict) -> BaseArtifact:
         path = self.path
         body = values["body"]
         url = self._build_url(base_url, path=path)
+        headers = values.get("headers", self.request_headers)
```

Comment:
`values["headers"]` will never be set since it is not described in the schema for the LLM to use. Though I don't know if we _want_ the LLM setting headers on the fly. Certainly not if they're going to override the user set headers.

What do you think? Do you have a use case in mind?

Reply:
@collindutter values isn't defined in the schema either. maybe I misunderstand how this is supposed to be used, but from my testing with this using the gitlab api, it seemed that there was a 1:1 relationship between RestApiClient and endpoint for the api. I talked about it in this comment from earlier in the PR:  https://github.com/griptape-ai/griptape/pull/535#issuecomment-1872518512

Reply:
the intention was to be able to give an agent a knowledge base about the available endpoints of the api and their usage, and a single RestApiClient that was instantiated for the base url for the api and then be able to optionally pass custom values for headers & path at runtime based on what it learns from the knowledge base.

Reply:
`values` is special field [added by Griptape](https://github.com/griptape-ai/griptape/blob/dev/griptape/mixins/activity_mixin.py#L81). Your use-case makes sense, though I fear the LLM would override user set headers.

Maybe it's just one or the other? Default to user set headers (via `self.headers`), and fall back to LLM set headers (via `values["headers"]`).

@vasinov what do you think?

Reply:
@collindutter thank you for the info about values. Also I see your point and agree about the headers, my main intention and goal was for being able to set the path at runtime.

Reply:
@andrewfrench what do you think?

Reply:
@mgcollie after discussing it with the team, I think we've landed on not allowing the LLM to set headers. These should always be explicitly set by the developer. So we just want:
```python
response = put(url, json=body, timeout=30, headers=self.request_headers)
```